src/extension.ts

```diff
@@ -10,6 +10,60 @@ import { getRefAt, RefType } from './Ref';
 // import { debug } from 'util';
 // import { create } from 'domain';
 
+let c = vscode.workspace.getConfiguration("vscodeMarkdownNotes");
+
+// Function that returns a filename based on the given wikilink.
+// Initially uses filesForWikiLinkRefFromCache() to try and find a matching file.
+// If this fails, it will attempt to make a (relative) link based on the label given.
+function PageNameGenerator(label: string) {
+    console.debug(label);
+    const ref = {
+        type: RefType.WikiLink,
+        word: label,
+        hasExtension: false,
+        range: undefined
+    }
+    const results = MarkdownDefinitionProvider.filesForWikiLinkRefFromCache(ref, null);
+
+    label = label.replace(/\.[^\.\\\/]+$/, '');
+
+    // Either use the first result of the cache, or in the case that it's empty use the label to create a path
+    let path: string = (results.length != 0) ? results[0].path : NoteWorkspace.noteFileNameFromTitle(label);
+    
+    return path;
+}
+
+function postProcessPageName(pageName: string) {
+    pageName = pageName.trim();
+    pageName = pageName.replace(/\.[^\.\\\/]+$/, '');
+
+    return pageName;
+
+}
+  
+function postProcessLabel(label: string) {
+	label = label.trim();
+	
+	// Remove filename extension
+	label = label.replace(/\.[^/.]+$/, "");
+    
+    
+    label = label.split(`${c.get('slugifyCharacter')}`).join(" ");
+    if (c.get('showFileExtensionInPreview')) {
```

Comment:
Please implement a wrapper func similar to `NoteWorkspace.slugifyChar()` (which we can stub in tests):

Reply:
:+1: